src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Tree/Assignment.cs

```diff
@@ -14,10 +16,12 @@ public ExprAssignment(Expr lhs, Expr rhs)
             Flags = EXPRFLAG.EXF_ASSGOP;
         }
 
-        public Expr LHS { get; set; }
+        public Expr LHS
+        {
+            get => _lhs;
+            set => Type = (_lhs = value).Type;
```

Comment:
Rect. This particular one can be changed in the visitor, so no luck there, unless it later turns out the visitor will never make an actual change.